### ISSUE TITLE
Red Mountain (Sonic) - back of mountain model visibile

### ISSUE THREAD

**Issue by Sonikko** [PiKeyAr/sadx_dreamcast#12]
In Red Mountain as Sonic (no issue with Knuckles's level, due to the different background mountains layout I guess?) there's really noticeable clipping happening between the setpieces and both the skybox and the background polygonal mountains at this certain spot:
![screenshot 2016-12-24 15 09 07](https://cloud.githubusercontent.com/assets/24740235/21467402/95b357b6-c9ec-11e6-9c90-b2c62326ce14.png)
![screenshot 2016-12-24 15 07 55](https://cloud.githubusercontent.com/assets/24740235/21467404/95b47bd2-c9ec-11e6-806e-10504e419cab.png)
![screenshot 2016-12-24 15 07 59](https://cloud.githubusercontent.com/assets/24740235/21467403/95b43b86-c9ec-11e6-9d39-6f4f3ca34947.png)
![screenshot 2016-12-24 15 08 13](https://cloud.githubusercontent.com/assets/24740235/21467405/95b5a3e0-c9ec-11e6-983b-9512e6b9bad3.png)
![screenshot 2016-12-24 15 08 40](https://cloud.githubusercontent.com/assets/24740235/21467406/95b64a3e-c9ec-11e6-8cfd-96947fbaf7ed.png)
Also, as you can clearly see, the back side of the model can be seen.

Screens of Knuckles's section, same spots
![screenshot 2016-12-24 15 19 02](https://cloud.githubusercontent.com/assets/24740235/21467411/c1015dc8-c9ec-11e6-9c31-691333ecccff.png)
![screenshot 2016-12-24 15 19 09](https://cloud.githubusercontent.com/assets/24740235/21467412/c102ba24-c9ec-11e6-95b1-ca0365fa0151.png)

Not using any conflicting mods.
![image](https://cloud.githubusercontent.com/assets/24740235/21467420/fbaa8b70-c9ec-11e6-9f87-faf273c2ef28.png)
Better SADX is just ADX audio.
Running the latest ModLoader version.

**Comment by PiKeyAr** [comment]
Increasing draw distance for this level is about the only thing I can do about this. A side effect of that will be you being able to see really far, which isn't the same as on the Dreamcast, but maybe it's not a bad thing.

You can download an updated Red Mountain mod [from here](https://www.dropbox.com/sh/zhlvlswm08bh9ap/AACzfxoaK-E2trdBKGi1Ddrwa?dl=0), let me know if it fixes this (or breaks anything else).

**Comment by Sonikko** [comment]
The clipping issue is fixed, I tried the whole level and nothing seems to be broken by the different draw distance. I must say I think that it is quite the improvement. The back of the mountain (circled in red) remains visible, I think it needs to be remodeled to be actually fixed. Anyway, nice job. A draw distance edit for all the levels would look great imho.
![screenshot 2016-12-24 16 56 42](https://cloud.githubusercontent.com/assets/24740235/21467820/c611a012-c9fa-11e6-9c4a-32536dd6dfa5.png)
![screenshot 2016-12-24 16 58 20](https://cloud.githubusercontent.com/assets/24740235/21467821/c627b208-c9fa-11e6-9898-eb879bb3d86f.png)
![screenshot 2016-12-24 16 58 54](https://cloud.githubusercontent.com/assets/24740235/21467822/c63b2c34-c9fa-11e6-999f-ccb61cda8b45.png)
![screenshot 2016-12-24 16 59 33](https://cloud.githubusercontent.com/assets/24740235/21467823/c63d9f3c-c9fa-11e6-8a6f-5a366d5c910d.png)
I must say, the draw distance fix looks lovely with the limit break mod by Morph/SonicFreak94 (shame I can't figure out how to turn the mod hud off and that some levels glitch out, like Emerald Coast)
![screenshot 2016-12-24 17 04 56](https://cloud.githubusercontent.com/assets/24740235/21467839/5c944242-c9fb-11e6-9495-f756a8f19051.png)
![screenshot 2016-12-24 17 06 01](https://cloud.githubusercontent.com/assets/24740235/21467840/5c958878-c9fb-11e6-88fb-23921fc8be80.png)

**Comment by PiKeyAr** [comment]
Yeah, it's just the way that model is. It's the same in SA1. This is what it looks like in the editor:

![untitled](https://cloud.githubusercontent.com/assets/23312549/21467951/76def9c0-ca42-11e6-92a8-61d93da26c76.png)

It looks like they forgot to add a few meshes to one side of the mountain. The other side is fine. I don't think I'll be fixing it on my own, but if you feel like remodeling it, feel free to have your own take on it. I can integrate it into DC Mods later and credit you. Either .sa1lvl or .obj will be fine.

I am actually applying draw distance tweaks to most levels, but I don't want to increase it too far as it can cause other issues. Luckily it worked well in this level.

**Comment by Sonikko** [comment]
I'll be happy to test the draw distance tweaks for you, just let me know. My mail is in my profile description.
I've used 3d modeling tools in the past, but I've never been good at it ^^'. If I'll get around to it I'll contact you!

**Comment by PiKeyAr** [comment]
You can test different draw distance with Cheat Engine and this table: https://www.dropbox.com/s/466lq0zo0px1d60/sadx_fog.CT?dl=0
Open the table with Cheat Engine, run SADX and go to the level you want to test, then open the sonic.exe process and change the DrawDist value.
I'll close this for now, since the first issue is fixed, and the second issue is a problem with original Dreamcast assets and requires remodeling them, which is something I'm not very good at.

**Comment by PiKeyAr** [comment]
It just occurred to me that a rather easy solution to the mountain problem is to move and/or rotate the whole thing. It's not in the playable area, so there should be no issues with doing so. I feel like it should be rather low priority so I don't know when exactly I'll be fixing this, though if you rotate it on your own I'll be happy to include it.

**Comment by PiKeyAr** [comment]
I tried to rotate the mountain, but it's modeled to wrap around the other level pieces, and it looks weird if I move it around. Rotating it also causes a similar problem with another mountain. Moving both of them far away from the playable area could work, but I feel like this is too much of a departure from the original assets, which isn't worth it. Maybe a better solution will show up someday. Closing this for now.
